typescript/sdk/src/consts/environments/mainnet.json

```diff
@@ -153,5 +153,22 @@
     "interchainAccountRouter": "0xe9E3444DDD80c50276c0Fcf316026f6d7fEc2c47",
     "merkleRootMultisigIsmFactory": "0x784b9D0f4eF9fb8444DfB5d24AB221C9D1A85395",
     "messageIdMultisigIsmFactory": "0xC4275763D7b621eb732847957012F1fb35C90BB8"
+  },
+  "solana": {
```

Comment:
I'm surprised that adding this doesn't cause a bunch of tests to fail. There's are a few places where functions in the SDK enumerate over the chains + contracts and then try to instantiate factories for everything. None of the new multi-protocol stuff has that problem but some old classes do.

That's why I did this for Kathy: https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/typescript/infra/scripts/helloworld/utils.ts#L74

Reply:
Funnily enough not having this caused more things to fail and adding it but accommodating its presence throughout our tooling was easier. Most of the usage of `filterAddressesExcludeProtocol` / `filterAddressesToProtocol` is to accommodate it.

The reason why adding it was necessary was that there are a number of places where we use the `multiProvider.intersect` to get rid of chains that don't have artifacts, but then it creates errors if we e.e.g want to do things like enroll a Solana foreign deployment remote router, which requires the ability to look up the domain ID from the chain name